sources/ecommerce-app/src/util/component.js

```diff
@@ -50,16 +51,19 @@ export function useHeader() {
 export function useCart({ onEmpty } = {}) {
   const dispatch = useDispatch();
   const cart = useSelector(state => state.products.cart);
+  const locale = useSelector(state => state.products).query.locale;
+  const currency = useSelector(state => state.products).query.currency;
   useEffect(
     () => {
-      if (cart == null) {
-        dispatch(fetchCart());
-      } else if (cart.items.length === 0) {
+      if (cart && cart.items.length === 0) {
         onEmpty && onEmpty();
       }
     },
-    [cart]
+    [cart, onEmpty]
```

Comment:
Is this safe? Are consumers of useCart memoizing onEmpty?

Reply:
using refs